lib/errors/error_handling.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require_dependency "#{Rails.root}/lib/errors/course_update_error_handling"
+
+module ErrorHandling
+  include CourseUpdateErrorHandling
+
+  def perform_error_handling_tasks(error, level, extra, course, optional_params)
+    return report_exception_sentry(error, level, extra) unless course.present?
```

Comment:
Doing this twice, either at the beginning or end, is a confusing control flow.

Reply:
I have refactored the course update part into `CourseUpdateErrorHandling` module. Also moved its sentry error reportinng part separately there.